pkg/controller/networkpolicy/validate.go

```diff
@@ -213,23 +214,16 @@ func (v *NetworkPolicyValidator) validateAntreaPolicy(curObj, oldObj interface{}
 	return reason, allowed
 }
 
-// validatePortRange validates if portRange is valid
-func (a *antreaPolicyValidator) validatePortRange(ingress, egress []secv1alpha1.Rule) error {
+// validatePort validates if ports is valid
+func (a *antreaPolicyValidator) validatePort(ingress, egress []secv1alpha1.Rule) error {
 	isValid := func(rules []secv1alpha1.Rule) error {
 		for _, rule := range rules {
-			for _, portRange := range rule.PortRanges {
-				if (portRange.Range.From != nil) != (portRange.Range.To != nil) {
-					return fmt.Errorf("inside portRange, `from` and `to` have be set at the same time")
+			for _, port := range rule.Ports {
+				if port.EndPort != nil && port.Port == nil {
+					return fmt.Errorf("if `endPort` is specified `port` must be specified")
 				}
-				if portRange.Range.From != nil && portRange.Range.To != nil {
-					if *portRange.Range.From > *portRange.Range.To {
-						return fmt.Errorf("inside portRange, `from` cannot be bigger than `to`")
-					}
-					for _, except := range portRange.Range.Except {
-						if except <= *portRange.Range.From || except >= *portRange.Range.To {
-							return fmt.Errorf("ports in portRange.except should be inside (portRange.From, portRange.To) exclusive")
-						}
-					}
+				if port.Port.Type == intstr.String && port.EndPort != nil {
```

Comment:
Do we also need to verify `endPort` is greater than `port` numericially?

Reply:
Good point.